### PR TITLE
Fixes in the verification flow

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Some context:
On May 2021 we had a huge issue with verification. There was a huge refactor that brought a bunch of issues that we ended up reverting. _Before_ reverting though we fixed a lot of bugs related to it. In the middle of those bugfixes there were some other bugfixes that were pre-existing! However, when we reverted we reverted everything so we lost those fixes.

Here's the PR from that time: https://github.com/valora-inc/wallet/pull/432

In this PR I'm bringing back two bugfixes that will help with the verification issues.

- The first one is calling a lock we were already using properly. We were using a lock in a section on the code but we were not actually `awaiting` it so it was useless. There was a delay there to "fix" the issue but it didn't really fix it. The lock should now work properly.
- The second one is adding another lock in the code that receives the verification codes. Without it, the UI glitches out a bunch and there may be other issues as well, since all the codes pasted try to get inserted in the same spot. This happens on Android where codes are auto-pasted. With this change, all codes should be pasted in a different position.

This fixes issues for new users, but we also need to run a migration to fix the redux store state for existing users that show up as verified when they are not.

### Other changes

N/A

### Tested

Manually on both OSes. I also modified the code a bit to send the three codes instantly to make sure they are handled correctly.

### How others should test

Try verifying in Android and iOS several times. Try all the edge cases you can think of, including but not limited to:
- Copying the same code several times.
- Pasting the codes very quickly.
- Asking codes to be resent.

### Related issues

- Part of #1906
This doesn't fix the issue completely, it fixes it for new users but we still need to fix the internal state for existing ones.

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/identity/verification.ts [hunk 1]
```diff
@@ -956,12 +968,8 @@ function* completeAttestation(
     `Completing code (${code.shortCode} ${codePosition}) for issuer: ${code.issuer}`
   )
 
-  // Make each concurrent completion attempt wait a sec for where they are relative to other codes
-  // to ensure `processingInputCode` has enough time to properly gate the tx. 0-index code
-  // will have 0 delay, 1-index code will have 1 sec delay, etc.
   if (shouldUseKomenci) {
-    yield delay(codePosition * 5000)
-    yield inputAttestationCodeLock.acquireAsync()
+    yield call([inputAttestationCodeLock, inputAttestationCodeLock.acquireAsync])
```
Comment (gnardini) [inline packages/mobile/src/identity/verification.ts:972]: The delay was necessary because the lock wasn't working as intended because it was not being called correctly. redux-saga can be non-intuitive at times :/

File: packages/mobile/src/identity/verification.ts [hunk 2]
```diff
@@ -666,24 +667,29 @@ export function attestationCodeReceiver(
       action.index
     )
 
-    const attestationInputStatus = yield select(attestationInputStatusSelector)
```
Comment (gnardini) [inline packages/mobile/src/identity/verification.ts:669]: This block of code wasn't actually changed, I just moved the `try` further up the flow and added a lock

**Review by jeanregisser — APPROVED** [review]
Thanks so much! 🥇 💯 

Hopefully in the next iteration, we'll have better test coverage.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue on Latest Android Internal Release Build & Test flight Release Build V1.27.0 and updated status in Test rails.

- User is able to complete phone number verification by entering 3-confirmation codes. 
- But on changing phone and starting verification again "Confirmation error" popup appears. [Bug link](https://github.com/valora-inc/wallet/issues/381)

**Test rail links:**
**Android 10:** https://valoraapp.testrail.io/index.php?/tests/view/12913

**iOS 14:** https://valoraapp.testrail.io/index.php?/tests/view/12975

**iOS 15:** https://valoraapp.testrail.io/index.php?/tests/view/13006

**Devices:** iPhone 13 mini(15.1.1), iPhone 12 (14.7.1), Google Pixel 2XL (11.0)
**cc:** @gnardini 
Thanks..!!
